### PR TITLE
Add mbedtls

### PR DESCRIPTION
**Pull request by chehefen**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,6 +7,7 @@ project(KinesisVideoProducerC LANGUAGES C)
 option(ADD_MUCLIBC "Add -muclibc c flag" OFF)
 option(BUILD_DEPENDENCIES "Whether or not to build depending libraries from source" ON)
 option(USE_OPENSSL "Use openssl as crypto library" ON)
+option(USE_MBEDTLS "Use openssl as crypto library" OFF)
```
Comment (zhiyua-git) [inline CMakeLists.txt]: use mbedTLS?

File: CMakeLists.txt [hunk 2]
```diff
@@ -63,10 +64,21 @@ if(BUILD_DEPENDENCIES)
 
   message(STATUS "Begin building dependencies.")
 
+  count_true(ENABLED_SSL_OPTION_COUNT
+    USE_OPENSSL
+    USE_MBEDTLS)
+
+  if(ENABLED_SSL_OPTION_COUNT GREATER "1")
```
Comment (zhiyua-git) [inline CMakeLists.txt]: If my input is just cmake -DUSE_MBEDTLS=TRUE .., would it fail? If it fails, is the error message clear enough to let me know that I need to disable openssl by -DUSE_OPENSSL=FALSE?

File: src/source/Common/Include_i.h [hunk 3]
```diff
@@ -13,9 +13,15 @@ extern "C" {
 ////////////////////////////////////////////////////
 // Project include files
 ////////////////////////////////////////////////////
+#if defined(KVS_USE_OPENSSL)
 #include <openssl/sha.h>
 #include <openssl/hmac.h>
 #include <openssl/evp.h>
+#elif defined(KVS_USE_MBEDTLS)
+#include "mbedtls/sha256.h"
+#include <mbedtls/md.h>
```
Comment (zhiyua-git) [inline src/source/Common/Include_i.h]: minor: shouldn't both be <mbedtls/*.h>?

File: src/source/Common/AwsV4Signer.c [hunk 4]
```diff
@@ -1070,3 +1024,100 @@ PCHAR getRequestVerbString(HTTP_REQUEST_VERB verb)
 
     return NULL;
 }
+
+#if defined(KVS_USE_OPENSSL)
```
Comment (MushMal) [inline src/source/Common/AwsV4Signer.c]: I strongly discourage usage of conditional compilations in this manner. 

Recommend using a separate header that hides the #ifdef under a macro and use those macros in the single code body. 

You only need to abstract two API calls. sha256 and hmac.

File: src/source/Common/AwsV4Signer.h [hunk 5]
```diff
@@ -78,6 +78,10 @@ extern "C" {
 // URI-encoded backslash value
 #define URI_ENCODED_FORWARD_SLASH               "%2F"
 
+#define SHA256_DIGEST_LENGTH                	32
```
Comment (MushMal) [inline src/source/Common/AwsV4Signer.h]: Very minor: line up the values.

File: src/source/Common/Include_i.h (hunk 3 above)
Comment (MushMal) [reply src/source/Common/Include_i.h]: Using wildcards is a bad idea

Comment (zhiyua-git) [reply src/source/Common/Include_i.h]: Didn't mean wildcard, I meant, shouldn't it be both 
#include "mbedtls/sha256.h"
#include "mbedtls/md.h"
or 
#include <mbedtls/sha256.h>
#include <mbedtls/md.h>

File: CMakeLists.txt (hunk 1 above)
Comment (chehefen) [reply CMakeLists.txt]: all our variables are capitalized though.

File: src/source/Common/AwsV4Signer.h (hunk 5 above)
Comment (chehefen) [reply src/source/Common/AwsV4Signer.h]: it does line up in clion.

File: CMakeLists.txt (hunk 1 above)
Comment (MushMal) [reply CMakeLists.txt]: +1 - capitalized is used everywhere

File: src/source/Common/Include_i.h (hunk 3 above)
Comment (MushMal) [reply src/source/Common/Include_i.h]: Oh, I see. Yeah, I think angle brackets are the right way to go here

**Pull request merged**